ddl/ddl_api.go

```diff
@@ -3561,11 +3561,11 @@ func checkColumnWithIndexConstraint(tbInfo *model.TableInfo, originalCol, newCol
 }
 
 func checkAutoRandom(tableInfo *model.TableInfo, originCol *table.Column, specNewColumn *ast.ColumnDef) (uint64, error) {
-	if tableInfo.GetPkName().L != originCol.Name.L {
-		return 0, nil
-	}
 	// Disallow add/drop actions on auto_random.
-	oldRandBits := tableInfo.AutoRandomBits
+	var oldRandBits uint64
+	if tableInfo.GetPkName().L == originCol.Name.L {
+		oldRandBits = tableInfo.AutoRandomBits
+	}
```

Comment:
In multiple column primary key tables, the `auto_random` must be `0` because `PkIsHandle == false`. Thus the `oldRandBits` is still `0`.